activiti-core-common/activiti-expression-language/src/main/java/org/activiti/core/el/JsonNodeELResolver.java

```diff
@@ -241,6 +241,14 @@ public Object getValue(ELContext context, Object base, Object property) {
 		return result;
 	}
 
+    /**
+     * Returns the {@link ObjectMapper} used internally to convert {@link List}
+     * properties. Subclasses may override this method to provide a specific one
+     */
+	protected ObjectMapper getObjectMapper() {
```

Comment:
Instead of having subclasses only to be able to override the object mapper, I'd prefer to inject it via the constructor: Object mapper becomes a mandatory constructor parameter for JsonNodeElResolver and JuelExpressionResolver:
```java
protected ELContext buildContext (Map<String, Object> variables) {
        return new ELContextBuilder()
            .withResolvers(
                arrayResolver(),
                listResolver(),
                mapResolver(),
                jsonNodeResolver(objectMapper),
                beanResolver()
            )
            .withVariables(variables)
            .build();
    }```

Reply:
I prefer composition over inheritance too, and in fact, it was my first approach. The problem is the expression:

`Context.getProcessEngineConfiguration().getObjectMapper().convertValue(resultNode, List.class);`

If we pass this to a constructor, the instance will keep whatever this returned when the instance was created.  The process engine configuration depends on ThreadLocal variables so I'm not sure the result will be the same in both cases.

We could fix this using a Supplier<ObjectMapper> but I thought it would be more readable this way.

Reply:
It makes sense. Let's keep it then, as there are technical constraints.